xclim/sdba/_adjustment.py

```diff
@@ -615,3 +615,157 @@ def extremes_adjust(
     adjusted: xr.DataArray = (transition * scen) + ((1 - transition) * ds.scen)
     out = adjusted.rename("scen").squeeze("group", drop=True).to_dataset()
     return out
+
+
+def otc_adjust(
+    Y: np.ndarray,
+    X: np.ndarray,
+    src: np.ndarray | None = None,
+    bin_width: np.ndarray | None = None,
+    bin_origin: np.ndarray | None = None,
+    numItermax: int | None = 100_000_000,
+):
+    """Optimal Transport Correction of the bias of X with respect to Y.
+
+    Parameters
+    ----------
+    Y : np.ndarray
+        Bias correction reference, target of optimal transport.
+    X : np.ndarray
+        Historical data to be corrected.
+    src : np.ndarray | None
+        If not None, X is still used to compute the optimal transport, but it is used to correct src.
```

Comment:
Am I understanding correctly: `src` is the same idea as having a `X1` in `dOTC`, but in this case, we consider `X` and `src` are in the same time period, so we don't apply special procedures as in dOTC?

Reply:
This is used by dOTC at
```
    # Map hist to sim
    yX1 = otc_adjust(
        X1,
        X0,
        src=yX0,
        bin_width=bin_width,
        bin_origin=bin_origin,
        numItermax=numItermax,
    )
```
Here `yX0` is the transportation of `Y0` to `X0`, so `src` and `X0` are virtually the same. The only potential difference is the quantity of data. This comes from `SBCK`, but it might be possible to do `otc_adjust(X1, yX0)`.

Reply:
ah right, I missed that, makes sense, thanks.